src/rdpc/analysesProcessor.ts

```diff
@@ -69,6 +115,50 @@ export const analysisStream = async function* ({
   }
 };
 
+export const aggregateSpecimensByDonorId = (
+  analyses: AnalysisWithSpecimens[]
+): SpecimensByDonors => {
+  const result = analyses.reduce<SpecimensByDonors>((acc, analysis) => {
+    const donorWithSpecimens = analysis.donors.reduce<SpecimensByDonors>(
+      (_acc, donor) => {
+        _acc[donor.donorId] = donor.specimens;
+        return _acc;
+      },
+      {}
+    );
+
+    Object.entries(donorWithSpecimens).forEach(([donorId, specimens]) => {
+      const existing = acc[donorId];
+      const mergedSpcimens =
+        existing === undefined
+          ? specimens
+          : mergedSpecimens(specimens, existing);
+      acc[donorId] = mergedSpcimens;
+    });
+    return acc;
+  }, {});
+
+  return result;
+};
+
+/**
+ * Removes duplicate specimens in merged arrays
+ * @param existingSpecimens
+ * @param toMerge
+ */
+export const mergedSpecimens = (
```

Comment:
Simple typo `merged`, lets call this `mergeSpecimen`

Reply:
This method seems pretty specific to this file, used in `aggregateSpecimensByDonorId()` and `mergeAllPagesSpecimensByDonorId()` only, does it need to be exported? Would we want it to be used external to this file?

Reply:
Was testing this function locally so forgot to remove `export`